src/actions/clusterActions.js

```diff
@@ -808,3 +807,56 @@ export function clusterCreateKeyPair(clusterId, keypair) {
       });
   };
 }
+
+/**
+ * Takes a nodePool object and tries to patch it.
+ * Dispatches NODEPOOL_PATCH on patch and NODEPOOL_PATCH_ERROR
+ * on error.
+ *
+ * @param {Object} cluster Cluster modification object
+ */
+export function nodePoolPatch(nodePool, payload) {
+  return function(dispatch, getState) {
+    const token = getState().app.loggedInUser.auth.token;
+    const scheme = getState().app.loggedInUser.auth.scheme;
+
+    // This is to get the cluster id.
+    // TODO I think we should normalize our store to avoid this
+    // hard-to-write-and-to-read queries
+    const clusters = getState().entities.clusters.items;
+    const nodePoolsClusters = getState().entities.clusters.nodePoolsClusters;
+
+    const cluster = nodePoolsClusters
+      .map(nodePoolCluster => {
+        return {
+          id: nodePoolCluster,
+          nodePools: clusters[nodePoolCluster].nodePools.map(np => np.id),
+        };
+      })
+      .filter(cluster => cluster.nodePools.some(np => np === nodePool.id));
+
+    const clusterId = cluster[0].id;
+
+    // Optimistic update.
+    dispatch({
+      type: types.NODEPOOL_PATCH,
+      nodePool,
+      clusterId,
+      payload,
+    });
+
+    return nodePoolsApi
+      .modifyNodePool(scheme + ' ' + token, clusterId, nodePool.id, payload)
+      .catch(error => {
+        // Undo update to store if the API call fails.
+        dispatch({
+          type: types.NODEPOOL_PATCH_ERROR,
```

Comment:
Does the user get any feedback when things go wrong? Or does the name just jump back? Was trying to find a flash error message or something like that, but not seeing it yet.

Reply:
It was in the `ViewEditName` component, I think it's cleaner to put this notifications here, so this is what I did